pkg/volume/plugins.go

```diff
@@ -131,26 +133,51 @@ type VolumePluginMgr struct {
 
 // Spec is an internal representation of a volume.  All API volume types translate to Spec.
 type Spec struct {
-	Name                   string
-	VolumeSource           api.VolumeSource
-	PersistentVolumeSource api.PersistentVolumeSource
-	ReadOnly               bool
+	Name             string
+	Volume           *api.Volume
+	PersistentVolume *api.PersistentVolume
+	ReadOnly         bool
+}
+
+// VolumeConfig contains any configuration item required by a plugin.  Config is passed to the volume plugin
+// in the ProbeVolumePlugins func, which allows configuration by the binary hosting the volume plugins.
+// Usage of this struct directly is possible but NewVolumeConfig is recommended because it will contain default values.
+type VolumeConfig struct {
+	// the default scrub pod is a template pod used by plugins when creating new Recyclers.
+	// The plugin is required to change the pod's VolumeSource to match the volume of the plugin and set any
+	// necessary information on the scrubber, such as path, server, or timeout.
+	PersistentVolumeRecyclerDefaultScrubPod *api.Pod
+	// the minimum ActiveDeadlineSeconds for an NFS scrubber pod
+	PersistentVolumeRecyclerMinTimeoutNfs int64
```

Comment:
If this is a generic struct, why does it have words like "NFS" and "HostPath" which are specific plugins?  I have just started in on this PR (sorry) but this is the first thing that caught me.